# **Exploratory** **Data** **Analysis**

Import library and datasets

In [72]:
import pandas as pd

In [73]:
df = pd.read_csv('basketball_hof.csv')
master_df = pd.read_csv('basketball_master.csv')

Check datasets for possible common variables.

In [74]:
df.head()

year      hofID               name     category
0  1959        NaN  Amos Alonzo Stagg  Contributor
1  1959        NaN      Charles Hyatt       Player
2  1959        NaN      Edward Hickox  Contributor
3  1959  mikange01       George Mikan       Player
4  1959        NaN      Hank Luisetti       Player

In [75]:
master_df.head()

bioID useFirst firstName middleName  ...  hsState hsCountry   deathDate race
0  abdelal01     Alaa      Alaa        NaN  ...       NJ       USA  0000-00-00    B
1  abdulka01   Kareem    Kareem        NaN  ...       NY       USA  0000-00-00    B
2  abdulma01    Mahdi     Mahdi        NaN  ...  PA / MD       USA  2011-11-18    B
3  abdulma02  Mahmoud   Mahmoud        NaN  ...       MS       USA  0000-00-00    B
4  abdulta01    Tariq     Tariq        NaN  ...      NaN       FRA  0000-00-00    B

[5 rows x 26 columns]

There are no similar variables in the two dataframes. This means the data set cannot be merged for analysis. Hence, the focus will be only on the hall of fame dataset

Check for null cells in each variable

In [76]:
 df.isnull().sum()

year          0
hofID       188
name          0
category      0
dtype: int64

hof1D has 188 null cells. This makes it unuseable and the dataframe does not provide enough information on what might be a replacement. Hence, the hofID column will be dropped from the dataframe.

In [77]:
df = df.drop(['hofID'], axis = 1)

Now check if it has effected.

In [78]:
df.isnull().sum()

year        0
name        0
category    0
dtype: int64

Check the number of cells in each variable

In [79]:
df.count()

year        328
name        328
category    328
dtype: int64

In [87]:
df['year'].unique()

array([1959, 1960, 1961, 1962, 1963, 1964, 1965, 1966, 1968, 1969, 1970,
       1971, 1972, 1973, 1974, 1975, 1976, 1977, 1978, 1979, 1980, 1981,
       1982, 1983, 1984, 1985, 1986, 1987, 1988, 1989, 1990, 1991, 1992,
       1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003,
       2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012])

**Over the time range, a total of 328 Hall of Fame inductions have been made from 1959 to 2012**

**Also, there was no induction made in 1967**





.

Check the total inductees under each category

In [91]:
df['category'].value_counts()

Player         157
Coach           90
Contributor     58
Referee         14
Team             9
Name: category, dtype: int64

**There are 157 players, 90 coaches, 57 contributors, 14 refrees and 9 teams inducted into the hall of fame over the time range**

In [81]:
df['year'].value_counts()

1959    17
1961    17
2012    12
1960    10
2011    10
2010    10
1992     9
1993     8
1979     7
1964     7
2008     7
2007     7
1991     7
1982     7
2003     7
1985     7
1998     7
1997     7
1995     7
1962     6
1996     6
2002     6
2004     6
2006     6
2000     6
1986     6
1983     6
1972     6
1980     6
1973     6
1984     5
1977     5
2009     5
1969     5
2005     5
1987     5
1999     5
1978     5
1994     5
1976     4
1975     4
1981     4
1990     4
1968     4
1965     4
1988     4
1974     3
2001     3
1971     3
1989     3
1963     3
1970     2
1966     2
Name: year, dtype: int64

**From the output above, it is deduced that the maximum inductees in a year is 17 while the minimum is 2.**